.github/workflows/nestjs.yml

```diff
@@ -0,0 +1,36 @@
+name: Nest.js ci/cd pipeline
+
+on:
+  push:
+    branches: ['master']
+  pull_request:
+    branches: ['master']
```

Comment:
Having both pull_request and push event handlers would start the pipeline, when PR is created and when PR is merged. We want the pipeline to run only when we merge - remove ' pull_request: branches: ['master']' trigger

Reply:
If you want to have a test workflow - you can configure it to run on pull_request